### PR TITLE
fix(remoteshell): stop rewriting the administrator's sshd_config

### PR DESCRIPTION
**Pull request by viniciusferrao**
On every deployment `remoteshell` strips the administrator's `X11Forwarding`
and `MaxStartups` lines from `/etc/ssh/sshd_config` and appends its own, so
local settings are gone after each install.

xCAT does not depend on the `MaxStartups` value, so the postscript now leaves
it alone. For `X11Forwarding`, where the running config has an active
`Include <dir>/*.conf` line it writes `<dir>/01-xcat.conf` and leaves
`sshd_config` untouched; otherwise it edits in place as before, so anything
older than OpenSSH 8.2 is unaffected. The directory is read from the file, and
anything it cannot resolve to a single directory falls back to the in-place
edit.

An administrator who sets `X11Forwarding` above the `Include` line still wins.
That takes a deliberate edit, and respecting it is the intent.

#5936 fixed this with a `Match all` block and was reverted in 3c95f2b79:
`MaxStartups` is illegal inside `Match`, so the file stopped parsing. Jarrod
Johnson's diagnosis was right; this takes a different route.

### CODE REVIEW

File: xCAT-test/unit/remoteshell_sshd_config.t [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    return;
+}
+
+sub read_file {
+    my ($path) = @_;
+    return '' unless -e $path;
+    open my $fh, '<', $path or die "open $path: $!";
+    local $/;
+    return <$fh>;
+}
```
Comment (dhilst) [inline xCAT-test/unit/remoteshell_sshd_config.t]: We can use File::Slurper for read/write_file

File: xCAT/postscripts/remoteshell [hunk 2]
```diff
@@ -54,22 +54,110 @@ if [ "$USEFLOWCONTROL" = "YES" ] || [ "$USEFLOWCONTROL" = "yes" ] || [ "$USEFLOW
...
+    mv -f "$xcat_tmp" "$xcat_dest" 2>/dev/null || { rm -f "$xcat_tmp"; return 1; }
+    return 0
+}
+
+# sshd keeps the first value it finds for a keyword, so a setting ahead of the
+# Include line or in an earlier fragment still wins over ours; log it.
+xcat_warn_if_overridden() {
+    xcat_kw=$1
+    xcat_own="$XCATSSHDDROPINDIR/$2"
+    if sed -n '/^[[:space:]]*[Ii][Nn][Cc][Ll][Uu][Dd][Ee][[:space:]]/q; p' /etc/ssh/sshd_config 2>/dev/null | \
```
Comment (dhilst) [inline xCAT/postscripts/remoteshell]: Using `/I` _ignore-case_ would make it more readable

**Review by dhilst — CHANGES_REQUESTED** [review]

**Comment by viniciusferrao** [comment]
@dhilst `/I` applied.

File::Slurper not added: it is not core, and xCAT-test/unit has no non-core dependencies today, so the suite would fail wherever it is not installed. Kept the small `read_file` helper.

Also added test coverage for the Include detection, which had none before.

Can you look again?

**Review by dhilst — APPROVED** [review]
